kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java

```diff
@@ -74,22 +78,32 @@ public AdaptiveBigQueryWriter(BigQuery bigQuery,
   public void performWriteRequest(PartitionedTableId tableId,
                                   List<InsertAllRequest.RowToInsert> rows,
                                   String topic) {
-    InsertAllRequest request = createInsertAllRequest(tableId, rows);
-    InsertAllResponse writeResponse = bigQuery.insertAll(request);
-    // Should only perform one schema update attempt; may have to continue insert attempts due to
-    // BigQuery schema updates taking up to two minutes to take effect
-    if (writeResponse.hasErrors()
-        && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
-      try {
-        schemaManager.updateSchema(tableId.getBaseTableId(), topic);
-      } catch (BigQueryException exception) {
-        throw new BigQueryConnectException("Failed to update table schema for: " + tableId.getBaseTableId(), exception);
+    InsertAllResponse writeResponse = null;
+    InsertAllRequest request = null;
+
+    try {
+      request = createInsertAllRequest(tableId, rows);
+      writeResponse = bigQuery.insertAll(request);
+      // Should only perform one schema update attempt; may have to continue insert attempts due to
+      // BigQuery schema updates taking up to two minutes to take effect
+      if (writeResponse.hasErrors()
+              && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
+        attemptSchemaUpdate(tableId, topic);
       }
+    } catch (BigQueryException e) {
+      if (isTableMissingSchema(e)) {
+        attemptSchemaUpdate(tableId, topic);
+      } else {
+        throw e;
+      }
+
+      // If the table was missing its schema, we never received a writeResponse
+      writeResponse = bigQuery.insertAll(request);
```

Comment:
If a table has no schema, it will raise a `BigQueryException` and the `writeResponse` will never be set in the try block so we need to retry inserting after adding a schema to the table